File: Enumeration/3371.Identify the Largest Outlier in an Array/Solution.py
class Solution:
    def getLargestOutlier(self, nums: List[int]) -> int:
        ns = [0] * 2010
        s = 0
        for v in nums:
            ns[v + 1000] += 1
            s += v
        ans = -1010
        for v in nums:
            if (s - v) % 2 != 0:
                continue
            t = (s - v) // 2
            if t < -1000 or t > 1000:
                continue
            if ns[t + 1000] >= 2 or (ns[t + 1000] >= 1 and t != v):
                ans = max(ans, v)
        return ans

File: Enumeration/3371.Identify the Largest Outlier in an Array/Solution.java
class Solution {
    public int getLargestOutlier(int[] nums) {
        int[] ns = new int[2010];
        int s = 0;
        for (int v : nums) {
            ns[v + 1000]++;
            s += v;
        }
        int ans = -1010;
        for (int v : nums) {
            if ((s - v) % 2 != 0) {
                continue;
            }
            int t = (s - v) / 2;
            if (t < -1000 || t > 1000) {
                continue;
            }
            if (ns[t + 1000] >= 2 || (ns[t + 1000] >= 1 && t != v)) {
                ans = Math.max(ans, v);
            }
        }
        return ans;
    }
}